File: backend/src/main/java/com/bookstore_backend/demo/service/OrderService.java
package com.bookstore_backend.demo.service;

import com.bookstore_backend.demo.entity.Order;
import com.bookstore_backend.demo.entity.OrderItem;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;
import java.util.Map;

public interface OrderService {
    boolean addOrder(Map<Object, Object> param);
    boolean addOrderOne(Map<Object, Object> param);
    List<Order> showAllOrders();
    List<Order> showOneOrder(Map<Object, Object> param);
    Order findOrder(Integer id);
    Order findOrderItems(Integer id);
    List<OrderItem> findOrderItemsByTime(@RequestBody Map<Object, Object> param);
    List<OrderItem> userFindOrderItemsByTime(@RequestBody Map<Object, Object> param);
    List<Order> findOrderByTime(@RequestBody Map<Object, Object> param);
    List<Order> userFindOrderByTime(@RequestBody Map<Object, Object> param);
}

File: frontend/src/services/orderService.js
import {postRequest, postRequest_v2} from "../utils/ajax";

export const addOrder = (data, callback) => {
    const url = 'http://localhost:8080/addOrder';
    postRequest(url, data, callback);
};

export const addOrderOne = (data, callback) => {
    const url = 'http://localhost:8080/addOrderOne';
    postRequest(url, data, callback);
};

export const showAllOrders = (data, callback) => {
    const url = 'http://localhost:8080/showAllOrders';
    postRequest(url, data, callback);
};

export const showOneOrder = (data, callback) => {
    const url = 'http://localhost:8080/showOneOrder';
    postRequest(url, data, callback);
};

export const findOneOrder = (id, callback) => {
    const data = {id: id};
    const url = 'http://localhost:8080/findOneOrder';
    postRequest_v2(url, data, callback);
};

export const findOrderItems = (id, callback) => {
    const data = {id: id};
    const url = 'http://localhost:8080/findOrderItems';
    postRequest_v2(url, data, callback);
};

export const findOrderItemsByTime = (data, callback) => {
    const url = 'http://localhost:8080/findOrderItemsByTime';
    postRequest(url, data, callback);
};

export const userFindOrderItemsByTime = (data, callback) => {
    const url = 'http://localhost:8080/userFindOrderItemsByTime';
    postRequest(url, data, callback);
};